Failing `TestCatalog_ResolvePath` test

<!-- 
1. Please search to see if an issue already exists for the bug you encountered.
2. For support requests, FAQs or "How to" questions, please use the GitHub Discussions section instead - https://github.com/projectdiscovery/nuclei/discussions or
3. Join our discord server at https://discord.gg/projectdiscovery and post the question on the #nuclei channel.
-->

<!-- ISSUES MISSING IMPORTANT INFORMATION MAY BE CLOSED WITHOUT INVESTIGATION. -->

### Nuclei version:
dev

### Current Behavior:
```
--- FAIL: TestCatalog_ResolvePath (0.00s)     --- FAIL: TestCatalog_ResolvePath/relative_path_with_second_param (0.00s)         catalog_test.go:212: ResolvePath() error = no such path found: deprecated-tls.yaml, wantErr false
```
### Expected Behavior:
Passing build test 

### Steps To Reproduce:
https://github.com/projectdiscovery/nuclei/actions/runs/4516727717/jobs/7955344390?pr=3466#step:6:20


### Anything else:
https://github.com/projectdiscovery/nuclei/pull/3466#issuecomment-1483836394

--- a/v2/pkg/catalog/aws/catalog.go
+++ b/v2/pkg/catalog/aws/catalog.go
@@ -6,7 +6,7 @@ import (
 	"errors"
 	"fmt"
 	"io"
-	"path/filepath"
+	"path"
 	"strings"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -129,13 +129,14 @@ func (c Catalog) ResolvePath(templateName, second string) (string, error) {
 
 	// if c second path is given, it's c folder and we join the two and check against keys
 	if second != "" {
-		target := filepath.Join(filepath.Dir(second), templateName)
+		// Note: Do not replace `path` with `filepath` since filepath is aware of Os path seperator
+		// and we only see `/` in s3 paths changing it to filepath cause build fail and other errors
+		target := path.Join(path.Dir(second), templateName)
 		for _, key := range keys {
 			if key == target {
 				return key, nil
 			}
 		}
-
 	}
 
 	// check if templateName is already an absolute path to c key

--- a/v2/pkg/catalog/aws/catalog.go
+++ b/v2/pkg/catalog/aws/catalog.go
@@ -135,6 +135,7 @@ func (c Catalog) ResolvePath(templateName, second string) (string, error) {
 				return key, nil
 			}
 		}
+
 	}
 
 	// check if templateName is already an absolute path to c key
@@ -144,7 +145,7 @@ func (c Catalog) ResolvePath(templateName, second string) (string, error) {
 		}
 	}
 
-	return "", fmt.Errorf("no such path found: %s", templateName)
+	return "", fmt.Errorf("no such path found: %s%s for keys: %v", second, templateName, keys)
 }
 
 func (s *s3svc) getAllKeys() ([]string, error) {

--- a/v2/pkg/catalog/aws/catalog_test.go
+++ b/v2/pkg/catalog/aws/catalog_test.go
@@ -1,11 +1,12 @@
 package aws
 
 import (
-	"github.com/pkg/errors"
 	"io"
 	"reflect"
 	"strings"
 	"testing"
+
+	"github.com/pkg/errors"
 )
 
 func TestCatalog_GetTemplatePath(t *testing.T) {

